homeassistant/components/switch/scsgate.py

```diff
@@ -0,0 +1,238 @@
+    """ Add only SCSGate scenario switches """
+    scenario = config.get("scenario")
+    switches = []
+
+    if scenario:
+        for _, entity_info in scenario.items():
+            if entity_info['scs_id'] in scsgate.SCSGATE.devices:
+                continue
+
+            logger.info(
+                "Adding %s scsgate.scenario_switch", entity_info['name'])
+
+            name = entity_info['name']
+            scs_id = entity_info['scs_id']
+
+            switch = SCSGateScenarioSwitch(
+                name=name,
+                scs_id=scs_id,
+                logger=logger)
+            switches.append(switch)
+            scsgate.SCSGATE.add_device(switch)
+    add_devices_callback(switches)
+
+
+class SCSGateSwitch(SwitchDevice):
+    """ Provides a SCSGate switch. """
+    def __init__(self, scs_id, name, logger):
+        self._name = name
+        self._scs_id = scs_id
+        self._toggled = False
+        self._logger = logger
+
+    @property
+    def scs_id(self):
+        """ SCS ID """
+        return self._scs_id
+
+    @property
+    def should_poll(self):
+        """ No polling needed for a SCSGate switch. """
+        return False
+
+    @property
+    def name(self):
+        """ Returns the name of the device if any. """
+        return self._name
+
+    @property
+    def is_on(self):
+        """ True if switch is on. """
+        return self._toggled
+
+    def turn_on(self, **kwargs):
+        """ Turn the device on. """
+        scsgate.SCSGATE.append_task(
+            ToggleStatusTask(
+                target=self._scs_id,
+                toggled=True))
+
+        self._toggled = True
+        self.update_ha_state()
+
+    def turn_off(self, **kwargs):
+        """ Turn the device off. """
+        scsgate.SCSGATE.append_task(
+            ToggleStatusTask(
+                target=self._scs_id,
+                toggled=False))
+
+        self._toggled = False
+        self.update_ha_state()
+
+    def process_event(self, message):
+        """ Handle a SCSGate message related with this switch"""
+        if self._toggled == message.toggled:
+            self._logger.info(
+                "Switch %s, ignoring message %s because state already active",
+                self._scs_id, message)
+            # Nothing changed, ignoring
+            return
+
+        self._logger.info(
+            "Switch %s, changing status from %s to %s",
+            self._scs_id, self._toggled, message.toggled)
+        self._toggled = message.toggled
+
+        self.update_ha_state()
+
+        command = "off"
+        if self._toggled:
+            command = "on"
+
+        self._logger.debug(
+            "Switch %s, firing state change %s",
+            self._scs_id, command)
+        self.hass.bus.fire(
+            'button_pressed', {
+                ATTR_ENTITY_ID: self._scs_id,
+                'state': command
+            }
+        )
+
+
+class SCSGateScenarioSwitch(SwitchDevice):
```

Comment:
This is a physical switch that have no on/off state. It's just a simple trigger. I decided to be coherent with the rest of the design. By default the device is hidden, so it doesn't clutter the UI.